src/hooks/useTrendingTopics.tsx

```
@@ -0,0 +1,53 @@
+  filters: Partial<SnippetFilters>
+  language: string
+  enabled?: boolean
+}) => {
+  return useQuery<TopicDetailsResponse, Error>({
+    queryKey: trendingKeys.topicDetails(topicId || '', timespan, filters, language),
+    queryFn: () => fetchTopicDetails({ topicId: topicId!, timespan, filters, language }),
+    staleTime: 1000 * 60 * 5, // Cache for 5 minutes
+    placeholderData: keepPreviousData, // Keep showing previous data while fetching
+    enabled: enabled && !!topicId
+  })
+}
```

ESLint

[error] Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`.

(arrow-body-style)